PyInstaller/compat.py

```diff
@@ -68,9 +70,10 @@
 # environment variable; which allows explicitly enabling or disabling the compatibility mode. However, note that
 # disabling the compatibility mode and using python that does not properly support Big Sur still leaves find_library()
 # broken (which is a scenario that we ignore at the moment).
-is_macos_11_compat = is_darwin and platform.mac_ver()[0].startswith('10.16')
-is_macos_11_native = is_darwin and platform.mac_ver()[0].startswith('11')
-is_macos_11 = is_macos_11_compat or is_macos_11_native
+# The same logic applies to macOS 12 (Monterey).
+is_macos_11_compat = _macos_ver and _macos_ver[0:2] == (10, 16)  # Big Sur or newer in compat mode
```

Comment:
Just to confirm, `platform.mac_ver()` gives 10.16 on both macOS 11 and 12 when in compat mode?

Reply:
Yeah.